pkg/kubelet/dockertools/manager.go

```diff
@@ -1826,6 +1828,13 @@ func (dm *DockerManager) SyncPod(pod *api.Pod, runningPod kubecontainer.Pod, pod
 		if len(containerChanges.ContainersToKeep) == 0 && len(containerChanges.ContainersToStart) == 0 {
 			glog.V(4).Infof("Killing Infra Container for %q because all other containers are dead.", podFullName)
 		} else {
+			if containerChanges.InfraChanged {
+				ref, err := api.GetReference(pod)
+				if err != nil {
```

Comment:
Yes. This err is an usual GetReference err, not precondition for event. I want to record an event if InfraChanged.